compiler-lib/src/type_checking/type_system.rs

```diff
@@ -48,21 +49,39 @@ impl<'src> TypeSystem<'src> {
         self.defined_classes.get_mut(&name)
     }
 
-    pub fn lookup_class(&self, name: Symbol<'src>) -> Option<&ClassDef<'src>> {
-        self.defined_classes.get(&name)
+    pub fn lookup_class(&self, name: Symbol<'src>) -> Option<(&ClassDef<'src>, ClassDefId<'src>)> {
```

Comment:
In the ir codegen, we invoke this like the following:

```
let class_type = self.type_system.lookup_class(class.data.name.data);
```

Would be nicer to just invoke it with `class.data`, where `class` has type 

```
 `&asciifile::spanned::Spanned<'_, ast::ClassDeclaration<'_>>`                                                                                                                                                         
```

Reply:
This is not possible, since we also need to be able to call this function with `ty.name` and other symbols.

```bash
$ rg lookup_class
compiler-lib/src/type_checking/method_body_type_checker.rs
333:            NewObject(name) => match self.type_system.lookup_class(name.data) {
568:            match self.type_system.lookup_class(var_name.data) {

compiler-lib/src/type_checking/checker.rs
82:        let name = match type_system.lookup_class_mut(class_decl.name.data) {
233:        Custom(name) => match type_system.lookup_class(*name) {
```

Reply:
This is what the `TypeAnalysis` struct is for:
`type_analysis.decl_get_class_id(class_decl: &'ast ast::ClassDeclaration<'src>) -> Option<ClassDefId<'src>>`.